packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -535,30 +535,60 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
   }
 
   void _initMouseGestureRecognizer() {
-    switch (defaultTargetPlatform) {
-      case TargetPlatform.android:
-      case TargetPlatform.fuchsia:
-      case TargetPlatform.iOS:
-      case TargetPlatform.linux:
-      case TargetPlatform.macOS:
-      case TargetPlatform.windows:
-        _gestureRecognizers[TapAndPanGestureRecognizer] = GestureRecognizerFactoryWithHandlers<TapAndPanGestureRecognizer>(
-              () => TapAndPanGestureRecognizer(debugOwner:this),
-              (TapAndPanGestureRecognizer instance) {
-            instance
-              ..onTapDown = _startNewMouseSelectionGesture
-              ..onTapUp = _handleMouseTapUp
-              ..onDragStart = _handleMouseDragStart
-              ..onDragUpdate = _handleMouseDragUpdate
-              ..onDragEnd = _handleMouseDragEnd
-              ..onCancel = _clearSelection
-              ..dragStartBehavior = DragStartBehavior.down;
-          },
-        );
-    }
+    _gestureRecognizers[TapAndPanGestureRecognizer] = GestureRecognizerFactoryWithHandlers<TapAndPanGestureRecognizer>(
+          () => TapAndPanGestureRecognizer(
+            debugOwner:this,
+            supportedDevices: <PointerDeviceKind>{ PointerDeviceKind.mouse },
+          ),
+          (TapAndPanGestureRecognizer instance) {
+        instance
+          ..onTapDown = _startNewMouseSelectionGesture
+          ..onTapUp = _handleMouseTapUp
+          ..onDragStart = _handleMouseDragStart
+          ..onDragUpdate = _handleMouseDragUpdate
+          ..onDragEnd = _handleMouseDragEnd
+          ..onCancel = _clearSelection
+          ..dragStartBehavior = DragStartBehavior.down;
+      },
+    );
   }
 
   void _initTouchGestureRecognizer() {
+    // A [TapAndHorizontalDragGestureRecognizer] is used on non-precise pointer devices
+    // like PointerDeviceKind.touch so [SelectableRegion] gestures do not conflict with
+    // ancestor Scrollable gestures in common scenarios like a vertically scrolling list view.
+    _gestureRecognizers[TapAndHorizontalDragGestureRecognizer] = GestureRecognizerFactoryWithHandlers<TapAndHorizontalDragGestureRecognizer>(
```

Comment:
does the touch gesture support drag selection without tapping the selection gesture?

Reply:
Do you mean a single tap followed by a drag? Touch devices do not support this only mouse. There is an existing check that looks for this https://github.com/flutter/flutter/blob/b9e4416ad358892a24fb4f50a56ce1b0b3a54260/packages/flutter/lib/src/widgets/selectable_region.dart#L632-L636 and https://github.com/flutter/flutter/blob/b9e4416ad358892a24fb4f50a56ce1b0b3a54260/packages/flutter/lib/src/widgets/selectable_region.dart#L644-L648 .

Reply:
will using this gesture recognizer causes weirdness when user tap and vertical select? Base on reading the doc it looks like it would only accept horizontal drag first before winning the arena.

If the issue is with the slop, could we fix the vertical conflict by increase or decrease the tapandPanGestureRecognizer as well?

Reply:
It shouldn't cause weirdness with tap and vertical select, for example in the case of a double tap + vertical drag on Android or iOS, the `TapAndHorizontalDragGestureRecognizer` will declare victory on a consecutive tap https://github.com/flutter/flutter/blob/22a5c6cb0a906c219e360ab72da0935aa7a18b93/packages/flutter/lib/src/gestures/tap_and_drag.dart#L1322-L1327, so when the drag happens the recognizer has already won and will accept the drag in any direction.

I think the more we increase the slop the less responsive the selection gesture will feel. I don't think this is ideal since selection gestures will not always be used in a Scrollable. I explored this method with `TextField`s when fixing a similar issue and found it to be fragile.

Reply:
sounds good to me then.